packages/console/src/pages/SignInExperience/components/Preview.tsx

```diff
@@ -43,19 +44,48 @@ const Preview = ({ signInExperience, className }: Props) => {
       []
     );
 
-    return encodeURIComponent(
-      JSON.stringify({
-        signInExperience: {
-          ...signInExperience,
-          socialConnectors,
-        },
-        language,
-        mode,
-        platform,
-      })
-    );
+    return {
+      signInExperience: {
+        ...signInExperience,
+        socialConnectors,
+      },
+      language,
+      mode,
+      platform,
+    };
   }, [allConnectors, language, mode, platform, signInExperience]);
 
+  const postPreviewMessage = useCallback(() => {
+    if (!config) {
+      return;
+    }
+
+    previewRef.current?.contentWindow?.postMessage(
+      { sender: 'ac_preview', config },
+      window.location.origin
+    );
+  }, [config]);
+
+  useEffect(() => {
+    const onIFrameLoadHandler = (event: MessageEvent) => {
+      if (event.origin !== window.location.origin) {
+        return;
+      }
+
+      if (event.data.sender === 'logto_ui' && event.data.pageReady) {
+        postPreviewMessage();
+      }
+    };
+
+    postPreviewMessage();
```

Comment:
Should it be called at this line as well?

Reply:
yes, on config change should call postMessage to the openWindow

Reply:
But how? This useEffect doesn't seem to be designed to call multiple times, as there is only one static dependency. So how could it postMessage to "the openWindow" on config change?
And btw, does openWindow refer to the iFrame window?